### ISSUE TITLE
`max-depth` rule

### ISSUE THREAD

**Issue by SamVerschueren** [SamVerschueren/clinton#19]
`max-depth` would ensure that the maximum directory depth is not more than X. For instances, if we have the following project structure

```
- /
  - index.js
  - lib
     - utils
        - foo.js
```

Thus `foo.js` is located at depth is `3` and the rule will fail if `max-depth` is set to `2`.

Not sure if this is something we actually want. But the higher the depth, the more complex the project structure becomes. So it might be useful. And if so, wat would we use as value and should it be an error or a warning?

// @sindresorhus @jamestalmage @kevva @jfmengels

**Comment by sindresorhus** [comment]
I would do warning and depth of 4 as default.

**Comment by SamVerschueren** [comment]
Added this one but took `5` as default. I ran it over `AVA` and it was `8`. We can always reconsider the need of this rule in the future.
